src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/DateTimeVariantTest.cs

```diff
@@ -219,474 +223,402 @@ private static void TestSqlDataRecordParameterToTVP_Type(object paramValue, stri
+                conn.Open();
+                DropType(conn, tvpTypeName);
+                xsql(conn, string.Format("create type dbo.{0} as table (f1 sql_variant)", tvpTypeName));
 
-                    // Send TVP using SqlMetaData.
-                    SqlMetaData[] metadata = new SqlMetaData[1];
-                    metadata[0] = new SqlMetaData("f1", SqlDbType.Variant);
-                    SqlDataRecord[] record = new SqlDataRecord[1];
-                    record[0] = new SqlDataRecord(metadata);
-                    record[0].SetValue(0, paramValue);
+                // Send TVP using SqlMetaData.
+                SqlMetaData[] metadata = new SqlMetaData[1];
+                metadata[0] = new SqlMetaData("f1", SqlDbType.Variant);
+                SqlDataRecord[] record = new SqlDataRecord[1];
+                record[0] = new SqlDataRecord(metadata);
+                record[0].SetValue(0, paramValue);
 
-                    using (SqlCommand cmd = conn.CreateCommand())
-                    {
-                        cmd.CommandText = "select f1, sql_variant_property(f1,'BaseType') as BaseType from @tvpParam";
-                        SqlParameter p = cmd.Parameters.AddWithValue("@tvpParam", record);
-                        p.SqlDbType = SqlDbType.Structured;
-                        p.TypeName = string.Format("dbo.{0}", tvpTypeName);
-                        using (SqlDataReader dr = cmd.ExecuteReader())
-                        {
-                            dr.Read();
-                            VerifyReaderTypeAndValue("Test SqlDataRecord Parameter To TVP [Variant Type]", "SqlDbType.Variant", dr, expectedTypeName, expectedBaseTypeName, paramValue);
-                            dr.Dispose();
-                        }
-                    }
-                }
+                using SqlCommand cmd = conn.CreateCommand();
+                cmd.CommandText = "select f1, sql_variant_property(f1,'BaseType') as BaseType from @tvpParam";
+
+                SqlParameter p = cmd.Parameters.AddWithValue("@tvpParam", record);
+                p.SqlDbType = SqlDbType.Structured;
+                p.TypeName = string.Format("dbo.{0}", tvpTypeName);
+
+                using SqlDataReader dr = cmd.ExecuteReader();
+                dr.Read();
+
+                VerifyReaderTypeAndValue("Test SqlDataRecord Parameter To TVP [Variant Type]", "SqlDbType.Variant", dr, expectedTypeName, expectedBaseTypeName, paramValue);
+                dr.Dispose();
             }
             catch (Exception e)
             {
-                if (IsExpectedException(e, paramValue, expectedTypeName, expectedBaseTypeName))
-                    LogMessage(tag, "[EXPECTED EXCEPTION] " + e.Message);
-                else
-                    DisplayError(tag, e);
+                Assert.True(IsExpectedException(e, paramValue, expectedTypeName, expectedBaseTypeName), $"Expected exception did not happen. Current exception is: {e.Message}");
```

Comment:
Same with this one, it's never called.

Reply:
Yes,` IsExpected` needs to go away. I have tried in several places to adjust it. There is one part that is totally commented out and that part was called several times, but because there is no definition for `datetime2` and `timeoffset` in that function it makes frequent failures. It will be fixed completely in next phase.